catboost/python-package/catboost/core.py

```diff
@@ -3049,7 +3049,11 @@ def _tune_hyperparams(self, param_grid, X, y=None, cv=3, n_iter=10, partition_ra
                 search_by_train_test_split, calc_cv_statistics, custom_folds, verbose
             )
 
-        self.set_params(**cv_result['params'])
+        if not self.is_fitted():
+            self.set_params(**cv_result['params'])
+        else:
+            raise CatBoostError("Model was fitted before hyperparameters tuning. You can't change hyperparameters of fitted model.")
```

Comment:
We don't need to set params if we not to refit our model. So let's do like this:
```
if refit:
    if not self.is_fitted():
        raise CatBoostError("Model was fitted before hyperparameters tuning. You can't change hyperparameters of fitted model.")
    self.set_params(**cv_result['params'])
    self.fit(X, y, silent=True)
```